Automattic/wp-calypso
# Add migration key integration

Related to #91780
Based off of https://github.com/Automattic/wp-calypso/pull/92557

## Proposed Changes

* Integrate the migration key.

## Testing Instructions



* Create another WP site to migrate from.
* Create a new post in this site to validate the migration.
* Make sure you have the `migration-flow/new-migration-instructions-step` feature flag enabled.
* Create a new site with the Import intent.
* Navigate until the migration instructions step.
* Follow the steps.
* Make sure you can use the migration key and the migration works properly.

## Recording

https://github.com/user-attachments/assets/60ac72f2-4761-4ded-b7c0-ae6899cd6be5

Video starting the migration from another site:

https://github.com/user-attachments/assets/37954f86-7d55-418d-83f3-7ad13a0322b5

## Screenshots

<img width="445" alt="Screenshot 2024-07-18 at 11 39 30" src="https://github.com/user-attachments/assets/9e9248ad-e05d-4aa7-a537-1faf9e55333f">

<img width="442" alt="Screenshot 2024-07-18 at 11 40 26" src="https://github.com/user-attachments/assets/eb48c719-3c25-4899-8958-d9cd69ceffcf">

<img width="441" alt="Screenshot 2024-07-18 at 11 38 45" src="https://github.com/user-attachments/assets/5700b9b8-aaf6-48a5-acc2-3221e36ad304">


## Pre-merge Checklist
